src/averageAges.js

```diff
@@ -37,7 +48,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomen = people.filter(person =>
+    person.sex === 'f'
+    && (!withChildren || people.some(man => man.mother === person.name)));
+
+  const avgAge = filteredWomen.reduce((total, women) => {
```

Comment:
check variable name, don't make it shortened, use full naming for better understanding